lib/samples/identify_raster_cell/identify_raster_cell.dart

```diff
@@ -0,0 +1,148 @@
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // Raster layer to display raster data on the map.
+  late RasterLayer _rasterLayer;
+  // Graphic to get the raster cell information.
+  final Graphic _textGraphic = Graphic();
+  // Graphics overlay to display the text graphic.
+  final _textGraphicsOverlay = GraphicsOverlay();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            ArcGISMapView(
+              controllerProvider: () => _mapViewController,
+              onMapViewReady: onMapViewReady,
+              onTap: onTap,
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Define a new map with Wgs84 Spatial Reference.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISOceans);
+    map.initialViewpoint = Viewpoint.fromCenter(
+      ArcGISPoint(
+        x: -34.1,
+        y: 18.6,
+        spatialReference: SpatialReference.wgs84,
+        z: 9,
+      ),
+      scale: 10000,
+    );
+    // Add map to the map view.
+    _mapViewController.arcGISMap = map;
+    // Download the raster file.
+    await downloadSampleData(['b5f977c78ec74b3a8857ca86d1d9b318']);
+    // Get the application documents directory.
+    final appDir = await getApplicationDocumentsDirectory();
+    // Create a Raster from the local tif file.
+    final raster = Raster.withFileUri(
+      Uri.file(
+        '${appDir.absolute.path}/SA_EVI_8Day_03May20/SA_EVI_8Day_03May20.tif',
+      ),
+    );
+    // Create a Raster Layer.
+    _rasterLayer = RasterLayer.withRaster(raster);
+    // Load the Raster Layer.
+    await _rasterLayer.load();
+    // Add the Raster Layer to the map.
+    map.operationalLayers.add(_rasterLayer);
+    // Set the viewpoint.
+    if (_rasterLayer.fullExtent != null) {
+      await _mapViewController.setViewpointGeometry(_rasterLayer.fullExtent!);
+    }
+    _textGraphicsOverlay.graphics.add(_textGraphic);
+    _mapViewController.graphicsOverlays.add(_textGraphicsOverlay);
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  Future<void> onTap(Offset position) async {
+    // Get the result for where the user hovered on the raster layer.
+    final identifyResult = await _mapViewController.identifyLayer(
+      _rasterLayer,
+      screenPoint: position,
+      tolerance: 1,
+    );
+    // Create a StringBuffer to display information to the user.
+    final stringBuffer = StringBuffer();
```

Comment:
This variable declaration should be moved inside the "if" statement, because it is not used anywhere else.